### PR TITLE
Consolidate public room search experience

### PR DESCRIPTION
**Pull request by gsouquet**
Fixes https://github.com/vector-im/element-web/issues/22846

## Checklist

* [x] Tests written for new code (and old code if feasible)
* [x] Linter and other CI checks pass
* [x] Sign-off given on the changes (see [CONTRIBUTING.md](https://github.com/matrix-org/matrix-react-sdk/blob/develop/CONTRIBUTING.md))

---
Here's what your changelog entry will look like:

## ✨ Features
 * Consolidate public room search experience ([\#9605](https://github.com/matrix-org/matrix-react-sdk/pull/9605)). Fixes vector-im/element-web#22846.

### CODE REVIEW

File: src/Rooms.ts [hunk 1]
```diff
@@ -29,15 +29,15 @@ import AliasCustomisations from './customisations/Alias';
...
         room.getCanonicalAlias(), room.getAltAliases(),
     );
 }
 
 // The various display alias getters should all feed through this one path so
 // there's a single place to change the logic.
-export function getDisplayAliasForAliasSet(canonicalAlias: string, altAliases: string[]): string {
+export function getDisplayAliasForAliasSet(canonicalAlias?: string | null, altAliases?: string[]): string | undefined {
     if (AliasCustomisations.getDisplayAliasForAliasSet) {
         return AliasCustomisations.getDisplayAliasForAliasSet(canonicalAlias, altAliases);
```
Comment (turt2live) [inline src/Rooms.ts:42]: this appears to be a new tsc --strict failure, fwiw

File: cypress/e2e/room-directory/room-directory.spec.ts [hunk 2]
```diff
@@ -88,15 +88,16 @@ describe("Room Directory", () => {
...
+        cy.get('.mx_SpotlightDialog [aria-label="Search"]').type("Unknown Room");
+        cy.get(".mx_SpotlightDialog .mx_SpotlightDialog_otherSearches_messageSearchText")
+            .should("contain", "can't find the room you're looking for");
+        cy.get(".mx_SpotlightDialog").percySnapshotElement("Room Directory - filtered no results");
 
-        cy.get('.mx_RoomDirectory_dialogWrapper [name="dirsearch"]').type("{selectAll}{backspace}test1234");
-        cy.contains(".mx_RoomDirectory_dialogWrapper .mx_RoomDirectory_listItem", name)
+        cy.get('.mx_SpotlightDialog [aria-label="Search"]').type("{selectAll}{backspace}test1234");
+        cy.contains(".mx_SpotlightDialog .mx_SpotlightDialog_result_publicRoomName", name)
             .should("exist").as("resultRow");
```
Comment (turt2live) [inline cypress/e2e/room-directory/room-directory.spec.ts:98]: we can drop the `as` if we're not using it?

Comment (turt2live) [reply cypress/e2e/room-directory/room-directory.spec.ts:98]: (we should probably be using it though?)

**Review by kittykat — APPROVED** [review]
Makes sense from a product POV

**Comment by gsouquet** [comment]
@turt2live Looks like percy's test need review, and approval too. I do not have an account and can't change those.

**Comment by t3chguy** [comment]
I've approved the diffs which look sane, though a bunch of auth modals seem to have changed size which is worrying, and the changed snapshots for the room directory do not include the modal wrapper which makes the diff less visually useful - once those are addressed I will approve the Percy diff

File: src/utils/rooms.ts [hunk 3]
```diff
@@ -35,146 +24,3 @@ export function privateShouldBeEncrypted(): boolean {
...
-    } else {
-        // This is a 3rd party protocol. Let's see if we can join it
-        const protocolName = protocolNameForInstanceId(protocols, instanceId);
-        const instance = instanceForInstanceId(protocols, instanceId);
-        const fields = protocolName
-            ? getFieldsForThirdPartyLocation(alias, protocols[protocolName], instance)
-            : null;
-        if (!fields) {
-            const brand = SdkConfig.get().brand;
-            throw new GenericError(
```
Comment (MadLittleMods) [inline src/utils/rooms.ts:135]: This PR removed all instances of `GenericError`, we could also remove the class in general.

Removed in https://github.com/matrix-org/matrix-react-sdk/pull/10500

**Pull request merged**